### PR TITLE
Piggy bank + can_shatter element + improvements

### PR DESCRIPTION
**Pull request by aidan-bowman**
## About the Pull Request

Saw [this cute lil PR](https://github.com/tgstation/tgstation/pull/81900) and thought I would port it real quick, then realized this would be a perfect opportunity to port a few things over.

PERSISTENCE ATOMIZED INTO #1653 

## Why It's Good For The Game

The piggy bank is a nice lil feature that implements a small persistent element to the economy; it gives high-achieving logistics officers something to do and lets us "smooth over" economy unbalances (money gets taken out of high-cash rounds, and put into low-cash rounds).

Everything else is needed code quality.

## Changelog

:cl:
add: Piggy banks! The logistics officer has a vault that contains a special little piggy bank.
code: Unit test for required map objects.
code: Element can_shatter added for shatter handling.
refactor: General code improvements
/:cl:

### CODE REVIEW

**Comment by checkraisefold** [comment]
make sure to grab sprites for the ceramic material ?

**Comment by TheDarkElites** [comment]
You need to stop putting "improvments" in your PRs as an excuse to randomly fix code unrelated to your PR because you can't be bothered to make a separate PR for it

**Comment by aidan-bowman** [comment]
> You need to stop putting "improvments" in your PRs as an excuse to randomly fix code unrelated to your PR because you can't be bothered to make a separate PR for it

oh i "need" to huh? hooonk mimimimi hoooooooonk mimimimimimi

**Comment by Tsurupeta** [comment]
separate this piece of shit into 3 unrelated things then I'll think about merging 2 of them

**Comment by aidan-bowman** [comment]
> separate this piece of shit into 3 unrelated things then I'll think about merging 2 of them

Most of the changes I made here rely on each other, I can't atomize them without at least one PR waiting on another (unideal)

**Comment by TheDarkElites** [comment]
> > separate this piece of shit into 3 unrelated things then I'll think about merging 2 of them
> 
> 
> 
> Most of the changes I made here rely on each other, I can't atomize them without at least one PR waiting on another (unideal)

Can shatter element dosent rely on piggy bank and improvements dont rely on anything.

So Improvements->CanShatter->piggybank

**Review by Tsurupeta — CHANGES_REQUESTED** [review]
F

File: code/__defines/unit_tests.dm [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+					GLOB.required_map_items[type] = new_value; \
+				} else { \
+					existing_value.total_amount += 1; \
+				}; \
+			}; \
+		}; \
+	} while (FALSE)
+#else
+#define REGISTER_REQUIRED_MAP_ITEM(min, max)
+#endif
```
Comment (Tsurupeta) [inline code/__defines/unit_tests.dm:29]: we dont even have the fucking unit test, why are you adding this thing in a piggybank PR?

Comment (aidan-bowman) [reply code/__defines/unit_tests.dm:29]: I added the unit test as well

File: code/_global_vars/lists/sound.dm [hunk 2]
```diff
@@ -160,16 +160,21 @@ GLOBAL_LIST_INIT(sfx_list, list(
...
-	)
-
+		'sounds/scp/1507/attack6.ogg',
+	),
+
+	SFX_RATTLE = list(
+		'sounds/items/rattle1.ogg',
+		'sounds/items/rattle2.ogg',
+		'sounds/items/rattle3.ogg',
+	),
```
Comment (Tsurupeta) [inline code/_global_vars/lists/sound.dm:179]: Why did you just randomly decide to switch 1 file with lists to /tg/ list shit standard??????? but not others??? IN A PIGGYBANK PR?

File: code/__defines/unit_tests.dm (hunk 1 above)
Comment (Tsurupeta) [reply code/__defines/unit_tests.dm:29]: Ok, im blind looked for it in other folder

File: code/_global_vars/lists/sound.dm (hunk 2 above)
Comment (aidan-bowman) [reply code/_global_vars/lists/sound.dm:179]: i saw an opportunity for code cleanup and i took it. although, yeah, i should probably make a separate PR doing this globally

**Pull request closed without merging**